File: Basic/Math/CheckArmstrong.java
public class CheckArmstrong {
    public static boolean isArmstrong(int n) {
        int p = String.valueOf(n).length();
        int total = 0;
        int original = n;

        while (n != 0) {
            int digit = n % 10;
            total += Math.pow(digit, p);
            n /= 10;
        }

        return total == original;
    }
}

File: Basic/Math/CheckArmstrong.cpp
#include <cmath>
#include <string>

class CheckArmstrong {
public:
    static bool isArmstrong(int n) {
        int power = static_cast<int>(std::to_string(n).size());
        int original = n, total = 0;
        while (n != 0) {
            total += static_cast<int>(std::pow(n % 10, power));
            n /= 10;
        }
        return total == original;
    }
};